TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/build.gradle

```diff
@@ -0,0 +1,78 @@
+plugins {
+    id 'base'
+    id 'de.undercouch.download' version '5.5.0'
+}
+
+def apiPort = 8099 // Not a commonly used
+def openapiUrl = "http://127.0.0.1:${apiPort}/openapi.json"
+def serverPidFile = "${buildDir}/fastapi.pid"
+def openapiOutFile = "${buildDir}/openapi.json"
+
+def pythonSources = fileTree(dir: projectDir, include: '**/*.py')
+def dependencyFiles = files(
+    "Pipfile",
+    "Pipfile.lock"
+).filter { it.exists() }
+def venv = layout.projectDir.dir(".venv")
+
+// Needed dereferenced pointer for configuration cache
+def projectDirRef = projectDir
+
+tasks.register('preparePipenv') {
+    inputs.files(dependencyFiles)
+    outputs.dir(venv)
+
+    doLast {
+        ['pipenv', 'install', '--dev'].execute(null, projectDirRef).waitFor()
+        logger.info("Ran pipenv install")
+    }
+}
+
+tasks.register('startApiServer') {
```

Comment:
Just realized that to download the openapi file we start the server, download it, and stop the api server. Is this operation relatively quick?

Reply:
Yup - its really quick, hardly noticeable compared to the configuration cache